internal/host/hostutil/host_utils.go

```diff
@@ -151,6 +152,85 @@ func GetDiskByInstallationPath(disks []*models.Disk, installationPath string) *m
 	return result.(*models.Disk)
 }
 
+// GetAcceptableDisksWithHints finds acceptable installation disks based on the RootDeviceHints
+//
+// This function implements the logic to find the installation disk following what's currently
+// supported by OpenShift, instead of *all* the supported cases in Ironic. The following link
+// points to the RootDeviceDisk translation done by the BareMetal Operator that is then sent to
+// Ironic:
+// https://github.com/metal3-io/baremetal-operator/blob/dbe8780ad14f53132ba606d1baec808997febe49/pkg/provisioner/ironic/devicehints/devicehints.go#L11-L54
+//
+// The logic is quite straightforward and the checks done match what is in the aforementioned link.
+// Some string checks require equality, others partial equality, whereas the int checks require numeric comparison.
+//
+// Ironic's internal filter process requires that all the disks have to fully match the RootDeviceHints (and operation),
+// which is what this function does.
+//
+// This function also filters out disks that are not elegible for installation, as we already know those cannot be used.
+func GetAcceptableDisksWithHints(disks []*models.Disk, hints *bmh_v1alpha1.RootDeviceHints) []*models.Disk {
+	if hints == nil {
```

Comment:
As it stands in the current implementation, hints will never be nil due to the check in `findInstallationDiskID` is the desired behaviour that findInstallationDiskID in the absence of hints should return an arbitrary disk ID or should it return an empty ID.

If the former, then should the check be removed from `findInstallationDiskID` and if the latter should this check be removed here?

Reply:
Actually, noted that a few lines below there is an eligibility check for the disk, which will only run if this condition is not triggered. As it stands, this function could return ineligible disks in the collection if there are no hints specified.

Reply:
> Is the desired behaviour that findInstallationDiskID in the absence of hints should return an arbitrary disk ID or should it return an empty ID?

For the existing BMAC use, it should return nothing. For the new use in this PR, it's easier if it returns the list of all disks. Hence the checks in both places.

> Actually, noted that a few lines below there is an eligibility check for the disk, which will only run if this condition is not triggered. As it stands, this function could return ineligible disks in the collection if there are no hints specified.

That's a good point, thanks.